src/cpufreq.c

```diff
@@ -132,34 +132,36 @@ static int cpufreq_read(void) {
     /* convert kHz to Hz */
     v.gauge *= 1000.0;
 
-    cpufreq_submit(i, "cpufreq", NULL, &v);
+    cpufreq_submit(cpu, "cpufreq", NULL, &v);
 
     if (report_p_stats) {
       /* Read total transitions for cpu frequency */
       snprintf(filename, sizeof(filename),
-               "/sys/devices/system/cpu/cpu%d/cpufreq/stats/total_trans", i);
+               "/sys/devices/system/cpu/cpu%d/cpufreq/stats/total_trans", cpu);
       if (parse_value_file(filename, &v, DS_TYPE_COUNTER) != 0) {
         ERROR("cpufreq plugin: Reading \"%s\" failed.", filename);
         continue;
       }
-      cpufreq_submit(i, "counter", "transitions", &v);
+      cpufreq_submit(cpu, "counter", "transitions", &v);
```

Comment:
Please don't use the `counter` type. Maybe define a new `DERIVE` type called "transitions".

Reply:
Why not? "counter" type is used widely enough.

Also, there is small problem with `DERIVE` type. Derive type is signed int64_t, while value in file (as provided by kernel) is unsigned uint64_t.  How to deal?

Reply:
> Why not? "counter" type is used widely enough.

No, it's not:

```
cheery:collectd octo$ egrep -o 'ABSOLUTE|COUNTER|DERIVE|GAUGE' src/types.db | sort | uniq -c
   1 ABSOLUTE
  12 COUNTER
 154 DERIVE
 163 GAUGE
cheery:collectd octo$ grep COUNTER src/types.db
counter                 value:COUNTER:U:U
arc_counts              demand_data:COUNTER:0:U, demand_metadata:COUNTER:0:U, prefetch_data:COUNTER:0:U, prefetch_metadata:COUNTER:0:U
arc_l2_bytes            read:COUNTER:0:U, write:COUNTER:0:U
mysql_qcache            hits:COUNTER:0:U, inserts:COUNTER:0:U, not_cached:COUNTER:0:U, lowmem_prunes:COUNTER:0:U, queries_in_cache:GAUGE:0:U
mysql_threads           running:GAUGE:0:U, connected:GAUGE:0:U, cached:GAUGE:0:U, created:COUNTER:0:U
```

As you can see, with the exception of the `counter` type, all uses of `COUNTER` are legacy types that are no longer in active use.

> Also, there is small problem with `DERIVE` type. Derive type is signed int64_t, while value in file (as provided by kernel) is unsigned uint64_t. How to deal?

Ignore it, just cast to `derive_t`. This metrics increases by 100 every second, meaning it will overflow a signed integer after *(2^63 - 1) / 100* seconds – 2.9 billion years.